src/daemon/database/actions.rs

```diff
@@ -669,6 +669,36 @@ pub fn db_mark_rebroadcastable_spend(
     Ok(())
 }
 
+pub fn db_cancel_unvault(
+    db_path: &PathBuf,
+    cancel_txid: &Txid,
+    spend_txid: &Txid,
+) -> Result<(), DatabaseError> {
+    db_exec(db_path, |tx| {
+        tx.execute(
+            "UPDATE vaults SET status = (?1), updated_at = strftime('%s','now'), cancel_txid = (?2) \
+             WHERE vaults.id IN (SELECT vault_id FROM presigned_transactions WHERE txid = (?3))",
+            params![VaultStatus::Canceling as u32, cancel_txid.to_vec(), spend_txid.to_vec()],
+        )
+        .map_err(|e| DatabaseError(format!("Updating vault to 'spending': {}", e.to_string())))?;
+
+        Ok(())
+    })
+}
```

Comment:
So yea as outlined previously don't copy the Spend behaviour on this one, but also if you can move the helper for marking it as canceling just next to the other ones, i think you can even use the routine there